build_resolvers/lib/src/resolver.dart

```diff
@@ -18,26 +19,59 @@ import 'package:path/path.dart' as native_path;
 // We should always be using url paths here since it's always Dart/pub code.
 final path = native_path.url;
 
+/// Implements [Resolver.libraries] and [Resolver.findLibraryByName] by crawling
+/// down from entrypoints.
+class PerActionResolver implements ReleasableResolver {
+  final ReleasableResolver _delegate;
+  final Iterable<AssetId> _entryPoints;
+
+  PerActionResolver(this._delegate, this._entryPoints);
+
+  @override
+  Stream<LibraryElement> get libraries async* {
```

Comment:
should we add the `canRead` check here?

Reply:
Added a TODO. Do you think we need it right away?

Reply:
I don't think its strictly necessary no